src/EditorFeatures/Core.Wpf/InlineRename/UI/SmartRename/SmartRenameViewModel.cs

```diff
@@ -155,51 +167,61 @@ private void FetchSuggestions(bool isAutomaticOnInitialization)
         if (_getSuggestionsTask.Status is TaskStatus.RanToCompletion or TaskStatus.Faulted or TaskStatus.Canceled)
         {
             var listenerToken = _asyncListener.BeginAsyncOperation(nameof(_smartRenameSession.GetSuggestionsAsync));
-            _cancellationTokenSource?.Dispose();
+            _cancellationTokenSource.Dispose();
             _cancellationTokenSource = new CancellationTokenSource();
             _getSuggestionsTask = GetSuggestionsTaskAsync(isAutomaticOnInitialization, _cancellationTokenSource.Token).CompletesAsyncOperation(listenerToken);
         }
     }
 
     private async Task GetSuggestionsTaskAsync(bool isAutomaticOnInitialization, CancellationToken cancellationToken)
     {
-        if (isAutomaticOnInitialization)
+        _threadingContext.ThrowIfNotOnUIThread();
```

Comment:
note: if thsi is already an async method, don't assert this, use a JTF switch to the actual  main thread.  it will be a no-op if not on the main thread.

Reply:
Good point, this can break WPF.
Can you recommend a pattern to use in case cancellationToken has been canceled?
```
        finally
        {
            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken); // might be canceled
            this.IsInPreparation = false;
        }
        ```